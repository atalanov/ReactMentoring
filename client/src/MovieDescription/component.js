// @flow
import React from 'react';

export default props => (
  <div>
    <h4>{props.name}</h4>
    <p>{props.jenre} ({props.year})</p>
  </div>
);

