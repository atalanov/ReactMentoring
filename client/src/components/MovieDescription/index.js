import React from 'react';

export default class MovieDescription extends React.PureComponent {
    render() {
        return (
          <div>
            <h4>{this.props.name}</h4>
            <p>{this.props.jenre} ({this.props.year})</p>
          </div>
        );
    }
}
