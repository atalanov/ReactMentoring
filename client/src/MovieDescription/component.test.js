import React from 'react';
import { shallow, render } from 'enzyme';
import MovieDescription from './component';

describe('MovieDescription tests', () => {
    test('Component MovieDescription renders correctly', () => {
        const component = shallow(<MovieDescription name="TestFilm" jenre="Horror" year="2018" />);
        expect(component).toMatchSnapshot();
    });
});
