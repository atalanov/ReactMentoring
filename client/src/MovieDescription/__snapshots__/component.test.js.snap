// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDescription tests Component MovieDescription renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    jenre="Horror"
    name="TestFilm"
    year="2018"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h4>
          TestFilm
        </h4>,
        <p>
          Horror
           (
          2018
          )
        </p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "TestFilm",
        },
        "ref": null,
        "rendered": "TestFilm",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Horror",
            " (",
            "2018",
            ")",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Horror",
          " (",
          "2018",
          ")",
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h4>
            TestFilm
          </h4>,
          <p>
            Horror
             (
            2018
            )
          </p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "TestFilm",
          },
          "ref": null,
          "rendered": "TestFilm",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Horror",
              " (",
              "2018",
              ")",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Horror",
            " (",
            "2018",
            ")",
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
